doc/number.md

```diff
@@ -0,0 +1,53 @@
+# Number
```

Comment:
The doc should use markdown formatting as opposed to html formatting to be consistent with existing Node.js docs.